ui/js/emailverification.js

```diff
@@ -0,0 +1,68 @@
+/*!
+ * Copyright 2015 Apereo Foundation (AF) Licensed under the
+ * Educational Community License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may
+ * obtain a copy of the License at
+ *
+ *     http://opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an "AS IS"
+ * BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+require(['jquery','oae.core'], function($, oae) {
+
+    // If the user already has a verified email address, we simply redirect them to the /me page
+    if (oae.data.me.email && oae.data.me.emailVerified) {
+        return oae.api.util.redirect().me();
+    }
+
+    var token = $.url().param('token');
+
+    // The user needs to be logged in before we can verify their token
+    if (oae.data.me.anon) {
+        // When only a single external institutional authentication strategy (`cas`, `googleApps`, `shibboleth`)
+        // is enabled, the Sign In button should send the user directly to the sign in page. Alternatively, the sign
+        // in dropdown in the top navigation should be opened when clicking the Sign In button
+        var enabledStrategies = oae.api.authentication.getEnabledStrategies();
+        var singleInstitutionalAuth = null;
+        if (_.keys(enabledStrategies).length === 1 && _.contains(['cas', 'googleApps', 'shibboleth'], _.keys(enabledStrategies)[0])) {
```

Comment:
Can you create constants that are accessible from `oae.api.authentication` that represent CAS, Google and Shib keys used here, rather than using them as hard-coded strings?

e.g., `oae.api.authentication.STRATEGY_CAS`